typescript/infra/src/core/check.ts

```diff
@@ -32,61 +39,54 @@ export interface ValidatorViolation extends CheckerViolation {
   };
 }
 
-export class AbacusCoreChecker extends AbacusAppChecker<
-  AbacusCore,
-  CoreConfig
+export class AbacusCoreChecker<
+  Networks extends ChainName,
+> extends AbacusAppChecker<
+  Networks,
+  AbacusCore<Networks>,
+  CoreConfig<Networks>
 > {
-  async check(
-    owners: Partial<Record<types.Domain, types.Address>>,
-  ): Promise<void> {
-    await Promise.all(
-      this.app.domainNumbers.map((domain: types.Domain) => {
-        const owner = owners[domain];
-        if (!owner) throw new Error('owner not found');
-        return this.checkDomain(domain, owner);
-      }),
+  async checkOwners(owners: ChainMap<Networks, types.Address>) {
```

Comment:
This method seems to do more than just check owners? Imo all checkers should have the same interface of just `check`?